internal/state/execution.go

```diff
@@ -310,29 +310,33 @@ func (blockExec *BlockExecutor) ApplyBlock(
 	return state, nil
 }
 
-func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) (types.VoteExtension, error) {
+func (blockExec *BlockExecutor) ExtendVote(ctx context.Context, vote *types.Vote) ([]byte, error) {
```

Comment:
Sure we can do that. I suppose if we make breaking changes to the `ExtendVote` request in future it won't be too much of an issue to change this API too.

Reply:
Although if we do that, shouldn't we do the same for the other `BlockExecutor` functions for consistency? This would then result in `VerifyVoteExtension`, for example, requiring 5 parameters, which starts looking like a bit of a code smell.

I'd advocate for just keeping the single `vote` parameter (in addition to `ctx`) for simplicity's sake. This method's only ever called in one other location anyways, and isn't directly called in any tests right now.

Reply:
Ah, good point. I think your reasoning of keeping consistency between the two methods makes total sense.